test/integration/copy.test.js

```diff
@@ -68,92 +73,302 @@ test('test initialization', function (tt) {
         });
     });
 
-    tt.test('env', function (t) {
+    tt.test('create first container', function (t) {
+        t.plan(2);
         cli.run(t, { args: '-d --name ' + nginxName + ' nginx' });
     });
+
+    tt.test('create second container', function (t) {
+        t.plan(2);
+        cli.run(t, { args: '-d --name ' + nginxName2 + ' nginx' });
+    });
 });
 
 
-test('copy a file out of container', function (tt) {
+
+/**
+ * Tests
+ */
+
+// TODO:
+//
+// Copy out:
+//   * of a running container:
+//      - single file (and permuted permisisons)
+//      - directory structure with sub directories and files (and permuted
+//        permissions)
+//
+//   * of a stopped container:
+//      - single file (and permuted permisisons)
+//      - directory structure with sub directories and files (and permuted
+//        permissions)
+//
+// Copy in:
+//   * of a running container:
+//      - single file (and permuted permissions)
+//        - permuted permisisons
+//        - permuted date
+//      - directory structure with sub directories and files (and permuted
+//        permissions)
+//
+//   * of a stopped container:
+//      - single file (and permuted permisisons)
+//      - directory structure with sub directories and files (and permuted
+//        permissions)
+
+
+
+test('copy a file out of running container', function (tt) {
+    tt.plan(7);
     var fnbase = '/var/tmp';
-    var fn = 'random.test1';
+    var fn = 'copyout.test';
     var ffn = fnbase + '/' + fn;
+    var hash;
 
-    // Create a file and get a checksum of it
-    tt.test('create test file', function (t) {
-        var inside = [
-            'dd if=/dev/urandom count=1024 bs=1024',
-            'tee ' + ffn,
-            '/native/usr/bin/sum -x sha1'
-        ].join('| \\\n');
-        var args = sprintf('exec %s bash -c "%s"', nginxName, inside);
-        cli.docker(args, onDocker);
-
-        function onDocker(err, stdout, stderr) {
-            t.ifErr(err);
-            console.log(stdout.trim());
-            t.end();
+    vasync.waterfall([
+        function (next) {
+            createCopyOutFile(tt, ffn, nginxName, function (err, sha1) {
+                tt.ifErr(err, 'creating copy out file');
+                hash = sha1;
```

Comment:
`hash` is declared at the start of the test function, but only used within this scope, could use `sha1` directly in the `test.comment` immediately following. Should it perhaps be tested against the hash of the source file?